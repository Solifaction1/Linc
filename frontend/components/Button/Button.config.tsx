import React from 'react'

type propsConfig = {
	buttonType?: 'primary' | 'secondary' | 'tertiary' | string
	size?: 'sm' | 'md' | 'lg' | string
	disabled?: boolean
	isLoading?: boolean
	className?: string
	handleClick?: React.MouseEventHandler<HTMLButtonElement>
	children?: React.ReactNode
	label?: string
	icon?: string
	color?: string
}

const defaultProps = {
	buttonType: 'primary',
	size: 'md',
	disabled: false,
	isLoading: false,
	className: '',
	children: undefined,
	label: '',
	icon: '',
	color: 'beige',
}
export type { propsConfig }
export { defaultProps }
