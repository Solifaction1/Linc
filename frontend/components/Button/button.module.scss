@import 'styles/vars';
@import 'styles/utilities';

.button {
	position: relative;

	border-radius: 50px;

	cursor: pointer;

	font-family: $font-family;

	&[disabled],
	&[disabled]:hover,
	&[disabled]::before {
		background: transparent;

		opacity: 0.5;
		cursor: not-allowed;
		pointer-events: none;
	}

	&.sm {
		padding: 0.8rem 1.2rem;
		font-size: $font-size-text-sm;
	}

	&.md {
		padding: 0.8rem 2rem;
		font-size: $font-size-text-md;
	}

	&.lg {
		padding: 1.3rem 2.7rem;
		font-size: $font-size-text-lg;
	}

	.label {
		font-weight: $font-weight-light;
	}

	&.primary {
		border: 1px solid $color-beige-dark;
		background: transparent;

		.label {
			color: $color-beige-dark;
		}

		&:hover {
			background: $color-beige-dark-opacity-25;

			.label {
				color: $color-beige-dark;
			}
		}

		&:active {
			background: $color-beige-dark;

			.label {
				color: $color-white;
			}
		}

		&.is-loading {
			@extend %loading-animation;

			&::after {
				border-color: $color-gray-tint-80;
			}
		}

		&.light {
			border: 1px solid $color-beige;

			.label {
				color: $color-beige;
			}

			&:hover {
				background: $color-beige-opacity-25;
			}

			&:active {
				border: 1px solid $color-gray;
				background: $color-beige;

				.label {
					color: $color-gray;
				}
			}
		}

		&.dark {
			border: 1px solid $color-gray;

			.label {
				color: $color-gray;
			}

			&:hover {
				background: $color-gray-opacity-25;
			}

			&:active {
				background: $color-gray;

				.label {
					color: $color-beige;
				}
			}
		}
	}
	&.submit {
		border: 1px solid $color-beige-dark;
		background: $color-beige-dark;

		.label {
			color:$color-white;
		}

		&:hover {
			background: $color-beige-dark-opacity-25;

			.label {
				color: $color-beige-dark;
			}
		}

		&:active {
			background: $color-beige-dark;

			.label {
				color: $color-white;
			}
		}

		&.is-loading {
			@extend %loading-animation;

			&::after {
				border-color: $color-gray-tint-80;
			}
		}

		&.light {
			border: 1px solid $color-beige;

			.label {
				color: $color-beige;
			}

			&:hover {
				background: $color-beige-opacity-25;
			}

			&:active {
				border: 1px solid $color-gray;
				background: $color-beige;

				.label {
					color: $color-gray;
				}
			}
		}

		&.dark {
			border: 1px solid $color-gray;

			.label {
				color: $color-gray;
			}

			&:hover {
				background: $color-gray-opacity-25;
			}

			&:active {
				background: $color-gray;

				.label {
					color: $color-beige;
				}
			}
		}
	}

	&.secondary {
		border: 1px solid $color-beige-dark;
		background: transparent;

		&.is-loading {
			@extend %loading-animation;

			&::after {
				border-color: $color-gray;
			}
		}

		&.blueGray {
			border: 1px solid $color-blue-gray-dark;

			.label {
				color: $color-blue-gray-dark;
			}

			&:hover {
				background: $color-blue-gray-dark-opacity-25;
			}

			&:active {
				background: $color-blue-gray-dark;

				.label {
					color: $color-blue-gray;
				}
			}
		}

		&.darkGreen {
			border: 1px solid $color-green;

			.label {
				color: $color-green;
			}

			&:hover {
				background: $color-green-opacity-25;
			}

			&:active {
				background: $color-green;

				.label {
					color: $color-green-dark;
				}
			}
		}

		&.lightGreen {
			border: 1px solid $color-green-dark;

			.label {
				color: $color-green-dark;
			}

			&:hover {
				background: $color-green-dark-opacity-25;
			}

			&:active {
				background: $color-green-dark;

				.label {
					color: $color-green;
				}
			}
		}

		&.beige {
			border: 1px solid $color-beige-dark;

			.label {
				color: $color-beige-dark;
			}

			&:hover {
				background: $color-beige-dark-opacity-25;
			}

			&:active {
				background: $color-beige-dark;

				.label {
					color: $color-beige;
				}
			}
		}

		&.lightBlue {
			border: 1px solid $color-blue-dark;

			.label {
				color: $color-blue-dark;
			}

			&:hover {
				background: $color-blue-dark-opacity-25;
			}

			&:active {
				background: $color-blue-dark;

				.label {
					color: $color-blue;
				}
			}
		}

		&.darkBlue {
			border: 1px solid $color-blue;

			.label {
				color: $color-blue;
			}

			&:hover {
				background: $color-blue-opacity-25;
			}

			&:active {
				background: $color-blue;

				.label {
					color: $color-blue-dark;
				}
			}
		}

		&.red {
			border: 1px solid $color-red-dark;

			.label {
				color: $color-red-dark;
			}

			&:hover {
				background: $color-red-dark-opacity-25;
			}

			&:active {
				background: $color-red-dark;

				.label {
					color: $color-red;
				}
			}
		}
	}

	&.tertiary {
		border: 1px solid $color-white;
		background: transparent;

		.label {
			color: $color-white;
		}

		&:hover {
			background: $color-white-opacity-25;
		}

		&:active {
			background: $color-white;

			.label {
				color: transparent;
			}
		}

		&.is-loading {
			@extend %loading-animation;

			&::after {
				border-color: $color-gray;
			}
		}
	}
}
