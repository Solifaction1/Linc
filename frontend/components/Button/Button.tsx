import React from 'react'
import { Icon } from '@templates/switch-icon.config'
import SwitchIcon from '@templates/SwitchIcon'

import { defaultProps, propsConfig } from './Button.config'
import styles from './button.module.scss'

const Button: React.FC<propsConfig> = (props: propsConfig) => {
	return (
		<button
			type={props.buttonType === 'submit' ? 'submit' : 'button'}
			className={`${styles.button} ${props.isLoading && styles.isLoading} ${
				styles[props.buttonType || 'primary']
			} ${styles[props.size || 'sm']} ${!props.label && styles.noLabel} ${
				props.className && props.className
			} ${props.color && styles[props.color]}`}
			disabled={props.disabled || props.isLoading}
			onClick={
				props.buttonType !== 'submit'
					? props.handleClick || ((e) => e.preventDefault())
					: undefined
			}
		>
			{props.children}
			<p className={styles.label}>{props.label && props.label}</p>
			{props.icon && (
				<SwitchIcon icon={props.icon ? props.icon : Icon.ANGLE_RIGHT} />
			)}
		</button>
	)
}

Button.defaultProps = defaultProps

export default Button
