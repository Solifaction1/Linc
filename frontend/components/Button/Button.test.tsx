import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

import Button from './Button'

describe('Button', () => {
	it('renders a Button', () => {
		const { container } = render(
			<Button disabled isLoading className="Button" label="label">
				Children
			</Button>
		)

		expect(container).toMatchSnapshot()

		const button = container.querySelector('button')
		expect(button).toBeVisible()
		expect(button?.getAttribute('class')).toContain('button')
		expect(button?.getAttribute('class')).toContain('Button')
		expect(button?.getAttribute('class')).toContain('isLoading')
		expect(button?.getAttribute('class')).toContain('primary')
		expect(button?.getAttribute('class')).toContain('md')

		expect(button?.getAttribute('disabled')).toContain('')
		expect(button?.getAttribute('type')).toContain('button')

		expect(button?.textContent).toEqual('Childrenlabel')
	})
})
