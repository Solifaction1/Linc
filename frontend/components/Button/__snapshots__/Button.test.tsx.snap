// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders a Button 1`] = `
<div>
  <button
    class="button isLoading primary md false Button beige"
    disabled=""
    type="button"
  >
    Children
    <p
      class="label"
    >
      label
    </p>
  </button>
</div>
`;
